wordpress-mobile/WordPress-iOS
# Jetpack Focus: Add analytics tracking for the Jetpack Site Creation overlay

Part of #19585

## Description
This PR tracks the site creation overlay events. 

## Testing Instructions

P.S: To be able to override remote feature flag values, you can apply the changes made in #19645 locally.

1. Open the app
2. Make sure the "Jetpack Features Removal Phase one" flag is enabled from the debug menu
3. Navigate to the Sites list
4. Tap plus button
5. Tap "Create WordPress.com site"
6. The overlay is displayed
7. Check logs for the event `remove_site_creation_overlay_displayed`
8. Make sure the event properties include phase and source and are of the correct values
9. Tap the CTA button
10. Check logs for the event `remove_site_creation_overlay_button_tapped`
11. Make sure the event properties include phase and source and are of the correct values
12. Tap the continue button
13. Check logs for the event `remove_site_creation_overlay_dismissed`
14. Make sure the event properties include phase, source, and dismissal type and are of the correct values
15. Display the overlay again
16. Tap the close button
17. Check logs for the event `remove_site_creation_overlay_dismissed`
18. Make sure the event properties include phase, source, and dismissal type and are of the correct values
19. Repeat the same for login and signup flows

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
